agent/rpc/peering/service.go

```diff
@@ -600,6 +627,32 @@ func (s *Server) getExistingOrCreateNewPeerID(peerName, partition string) (strin
 	return id, nil
 }
 
+func (s *Server) getExistingPeering(peerName, partition string) (*pbpeering.Peering, error) {
+	q := state.Query{
+		Value:          strings.ToLower(peerName),
+		EnterpriseMeta: *structs.NodeEnterpriseMetaInPartition(partition),
+	}
+	_, peering, err := s.Backend.Store().PeeringRead(nil, q)
+	if err != nil {
+		return nil, err
+	}
+
+	return peering, nil
+}
+
+// validatePeerRole enforces the following rule for an existing peering:
+// - if a peering already exists, it can only be used as an acceptor or dialer
+//
+// We define a DIALER as a peering that has server addresses (or a peering that is created via the Establish endpoint)
+// Conversely, we define an ACCEPTOR as a peering that is created via the GenerateToken endpoint
+func (s *Server) validatePeerRole(peering *pbpeering.Peering, allowedToDial bool) error {
+	if peering != nil && peering.ShouldDial() != allowedToDial {
+		return fmt.Errorf("cannot create peering with name: %q; there is already a peering with that name and different role", peering.Name)
```

Comment:
Can we reword the error message and function name? "role" is no longer a used term here

Reply:
what would be a better phrase here?

Reply:
Assuming you're going to end up adding more validations here the function could simply be `validatePeer`.

For error message maybe a branch to say "there is an existing peering expecting to be dialed" vs "there is already an established peering"?

Reply:
thanks for the suggestion! added it